File: Frontend/nimrag-frontend/src/types/spotify.ts
export interface SpotifyTrack {
  track_name: string
  artist: string
  album: string
  album_cover_url: string | null
  progress_ms: number
  duration_ms: number
}

export interface NowPlayingResponse {
  is_playing: boolean
  track: SpotifyTrack | null
  queue: SpotifyTrack[]
}

File: Backend/src/schemas/spotify.py
from pydantic import BaseModel


class SpotifyTrack(BaseModel):
    track_name: str
    artist: str
    album: str
    album_cover_url: str | None = None
    progress_ms: int
    duration_ms: int


class NowPlayingResponse(BaseModel):
    is_playing: bool
    track: SpotifyTrack | None = None
    queue: list[SpotifyTrack] = []


class SpotifyAuthStatusResponse(BaseModel):
    authenticated: bool
    auth_url: str | None = None
